maven-plugins/shade-extensions/src/main/java/io/helidon/build/maven/shade/HelidonServiceTransformer.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.maven.shade;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.jar.JarOutputStream;
+import java.util.stream.Stream;
+
+import org.apache.maven.plugins.shade.relocation.Relocator;
+import org.apache.maven.plugins.shade.resource.ReproducibleResourceTransformer;
+
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Maven Shade plugin custom transformer for merging Helidon service-registry files.
+ * Usage:
+ * <pre>{@code
+ * <plugin>
+ *    <groupId>org.apache.maven.plugins</groupId>
+ *    <artifactId>maven-shade-plugin</artifactId>
+ *    <version>3.5.1</version>
+ *    <executions>
+ *        <execution>
+ *            <phase>package</phase>
+ *            <goals>
+ *                <goal>shade</goal>
+ *            </goals>
+ *            <configuration>
+ *                <transformers>
+ *                    <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+ *                    <transformer implementation="io.helidon.build.maven.shade.HelidonServiceTransformer"/>
+ *                </transformers>
+ *            </configuration>
+ *        </execution>
+ *    </executions>
+ *    <dependencies>
+ *        <dependency>
+ *            <groupId>io.helidon.build-tools</groupId>
+ *            <artifactId>helidon-shade-extensions</artifactId>
+ *            <version>4.0.0-SNAPSHOT</version>
```

Comment:
This is definitely wrong - never use a snapshot version. Maybe just use a reference to helidon-build-tools-version or something like that (or check other maven plugins in this project how this is done)

Reply:
It's just a comment left there by Daniel as an example. I removed SNAPSHOT from the version.